lib/plug/router/utils.ex

```diff
@@ -67,6 +67,21 @@ defmodule Plug.Router.Utils do
   end
 
   @doc """
+  Builds a list of path param names and var match pairs that can bind to
+  dynamic path segment values. Excludes path segments with underscores.
+
+  ## Examples
+
+      iex> Plug.Router.Utils.build_path_params_match([:id])
+      [{"id", {:id, [], nil}}]
+  """
+  def build_path_params_match(vars) do
+    vars
+    |> Enum.map(fn(var) -> {Atom.to_string(var), Macro.var(var, nil)} end)
+    |> Enum.filter(fn({var, _macro}) -> String.at(var, 0) !== "_" end)
```

Comment:
We could rewrite this (maybe more idiomatically) as

```elixir
vars
|> Enum.map(&{Atom.to_string(&1), Macro.var(&1, nil)})
|> Enum.reject(&match?({"_" <> _var, _macro}, &1))
```

Wdyt? Too cryptic? The first capture (`Enum.map/2`) is ok with `fn` as well, but using `match?/2` and `reject/2` (instead of `filter/2`) looks a bit cleaner to me (than `String.at/2` plus the `!==` operator).